core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -561,16 +561,33 @@ package object matryoshka {
     *
     * @group algtrans
     */
-  def attributeAlgebraM[F[_]: Functor, M[_]: Functor, A](f: AlgebraM[M, F, A]):
-      AlgebraM[M, F, Cofree[F, A]] =
-    fa => f(fa ∘ (_.head)) ∘ (Cofree(_, fa))
+  object attributeAlgebraM {
+    def apply[T] = new PartiallyApplied[T]
+
+    final class PartiallyApplied[T] {
+      def apply[F[_]: Functor, M[_]: Functor, A]
+        (f: AlgebraM[M, F, A])
+        (implicit T: Birecursive.Aux[T, EnvT[A, F, ?]])
+          : AlgebraM[M, F, T] =
+        fa => f(fa ∘ (_.project.ask)) ∘ (a => EnvT((a, fa)).embed)
```

Comment:
Ah, given `attributeElgotM` I'm guessing it refers to the argument, so nevermind.